### ISSUE TITLE
Assert in isUnitClassMaxedOut

### ISSUE THREAD

**Issue by billw2012** [caveman2cosmos/Caveman2Cosmos#116]
CvGame.cpp 6166

![image](https://user-images.githubusercontent.com/1453936/64479671-12b8de00-d1b2-11e9-9d3c-182b132b833c.png)

I think this assert is only valid if unlimited nation units option is not enabled right?
The unit in question 'UNITCLASS_MANSA_SAKURA' has 414 instances, when global max is set to 1, so I guess something didn't work right there!

**Comment by raxo2222** [comment]
This unit is hero, so it always should be only one unit max.

**Comment by Thunderbrd** [comment]
Yeah, if you're getting that on a Hero then something indeed went wrong - perhaps a barbarian spawning gone wrong?  Someone messing with their game with WB?  A problem in the initial units given to a new Barbarian emerging Civilization assigned in Python?  Otherwise, I can't even theorize how that took place.
